diff -up rustc-1.34.1-src/src/llvm-emscripten/cmake/modules/LLVMProcessSources.cmake.omv~ rustc-1.34.1-src/src/llvm-emscripten/cmake/modules/LLVMProcessSources.cmake
diff -up rustc-1.34.1-src/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake.omv~ rustc-1.34.1-src/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake
--- rustc-1.34.1-src/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake.omv~	2019-05-08 22:46:35.109544780 +0200
+++ rustc-1.34.1-src/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake	2019-05-08 22:47:27.468006189 +0200
@@ -53,14 +53,16 @@ function(llvm_process_sources OUT_VAR)
   set(sources ${ARG_UNPARSED_ARGUMENTS})
   llvm_check_source_file_list( ${sources} )
   
-  # This adds .td and .h files to the Visual Studio solution:
-  add_td_sources(sources)
-  find_all_header_files(hdrs "${ARG_ADDITIONAL_HEADER_DIRS}")
-  if (hdrs)
-    set_source_files_properties(${hdrs} PROPERTIES HEADER_FILE_ONLY ON)
+  if( LLVM_ENABLE_IDE )
+    # This adds .td and .h files to the Visual Studio solution:
+    add_td_sources(sources)
+    find_all_header_files(hdrs "${ARG_ADDITIONAL_HEADER_DIRS}")
+    if (hdrs)
+      set_source_files_properties(${hdrs} PROPERTIES HEADER_FILE_ONLY ON)
+    endif()
+    set_source_files_properties(${ARG_ADDITIONAL_HEADERS} PROPERTIES HEADER_FILE_ONLY ON)
+    list(APPEND sources ${ARG_ADDITIONAL_HEADERS} ${hdrs})
   endif()
-  set_source_files_properties(${ARG_ADDITIONAL_HEADERS} PROPERTIES HEADER_FILE_ONLY ON)
-  list(APPEND sources ${ARG_ADDITIONAL_HEADERS} ${hdrs})
 
   set( ${OUT_VAR} ${sources} PARENT_SCOPE )
 endfunction(llvm_process_sources)
