diff -uraN rustc-1.90.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs omv-rustc-1.90.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs
--- rustc-1.90.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs	2025-09-14 17:05:11.000000000 +0200
+++ omv-rustc-1.90.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs	2025-10-25 21:16:32.438926646 +0200
@@ -684,6 +684,7 @@
         // reduction.
         } else if (self.is_gnu || self.sess.target.is_like_wasm) && !keep_metadata {
             self.link_arg("--gc-sections");
+            self.link_arg("-z,nostart-stop-gc");
         }
     }
 
@@ -903,6 +904,7 @@
             self.link_arg("--nxcompat");
         } else if self.is_gnu {
             self.link_args(&["-z", "noexecstack"]);
+            self.link_args(&["-z", "nostart-stop-gc"]);
         }
     }
 
@@ -1401,6 +1403,7 @@
 
     fn gc_sections(&mut self, _keep_metadata: bool) {
         self.link_arg("--gc-sections");
+        self.link_arg("-z,nostart-stop-gc");
     }
 
     fn optimize(&mut self) {
@@ -1545,6 +1548,7 @@
     fn gc_sections(&mut self, keep_metadata: bool) {
         if !keep_metadata {
             self.link_arg("--gc-sections");
+            self.link_arg("-z,nostart-stop-gc");
         }
     }
 
