diff -Naru rustc-1.97.1-src/compiler/rustc_target/src/target_features.rs rustc-1.97.1-src/compiler/rustc_target/src/target_features.rs
--- rustc-1.97.1-src/compiler/rustc_target/src/target_features.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/compiler/rustc_target/src/target_features.rs	2026-07-23 17:00:08.688517894 +0200
@@ -398,7 +398,6 @@
     ("amx-fp16", Unstable(sym::x86_amx_intrinsics), &["amx-tile"]),
     ("amx-int8", Unstable(sym::x86_amx_intrinsics), &["amx-tile"]),
     ("amx-movrs", Unstable(sym::x86_amx_intrinsics), &["amx-tile"]),
-    ("amx-tf32", Unstable(sym::x86_amx_intrinsics), &["amx-tile"]),
     ("amx-tile", Unstable(sym::x86_amx_intrinsics), &[]),
     ("apxf", Unstable(sym::apx_target_feature), &[]),
     ("avx", Stable, &["sse4.2"]),
diff -Naru rustc-1.97.1-src/library/std_detect/src/detect/arch/x86.rs rustc-1.97.1-src/library/std_detect/src/detect/arch/x86.rs
--- rustc-1.97.1-src/library/std_detect/src/detect/arch/x86.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/library/std_detect/src/detect/arch/x86.rs	2026-07-23 17:00:08.691742195 +0200
@@ -91,7 +91,6 @@
     /// * `"amx-avx512"`
     /// * `"amx-fp8"`
     /// * `"amx-movrs"`
-    /// * `"amx-tf32"`
     /// * `"f16c"`
     /// * `"fma"`
     /// * `"bmi1"`
@@ -227,8 +226,6 @@
     /// AMX-FP8 (Float8 Operations)
     @FEATURE: #[unstable(feature = "x86_amx_intrinsics", issue = "126622")] amx_movrs: "amx-movrs";
     /// AMX-MOVRS (Matrix MOVERS operations)
-    @FEATURE: #[unstable(feature = "x86_amx_intrinsics", issue = "126622")] amx_tf32: "amx-tf32";
-    /// AMX-TF32 (TensorFloat32 Operations)
     @FEATURE: #[unstable(feature = "apx_target_feature", issue = "139284")] apxf: "apxf";
     /// APX-F (Advanced Performance Extensions - Foundation)
     @FEATURE: #[unstable(feature = "avx10_target_feature", issue = "138843")] avx10_1: "avx10.1";
diff -Naru rustc-1.97.1-src/library/std_detect/src/detect/os/x86.rs rustc-1.97.1-src/library/std_detect/src/detect/os/x86.rs
--- rustc-1.97.1-src/library/std_detect/src/detect/os/x86.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/library/std_detect/src/detect/os/x86.rs	2026-07-23 17:00:08.694508950 +0200
@@ -214,7 +214,6 @@
                             __cpuid_count(0x1e_u32, 1);
 
                         enable(amx_feature_flags_eax, 4, Feature::amx_fp8);
-                        enable(amx_feature_flags_eax, 6, Feature::amx_tf32);
                         enable(amx_feature_flags_eax, 7, Feature::amx_avx512);
                         enable(amx_feature_flags_eax, 8, Feature::amx_movrs);
                     }
diff -Naru rustc-1.97.1-src/library/std_detect/tests/x86-specific.rs rustc-1.97.1-src/library/std_detect/tests/x86-specific.rs
--- rustc-1.97.1-src/library/std_detect/tests/x86-specific.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/library/std_detect/tests/x86-specific.rs	2026-07-23 17:00:08.697330875 +0200
@@ -76,7 +76,6 @@
     println!("widekl: {:?}", is_x86_feature_detected!("widekl"));
     println!("movrs: {:?}", is_x86_feature_detected!("movrs"));
     println!("amx-fp8: {:?}", is_x86_feature_detected!("amx-fp8"));
-    println!("amx-tf32: {:?}", is_x86_feature_detected!("amx-tf32"));
     println!("amx-avx512: {:?}", is_x86_feature_detected!("amx-avx512"));
     println!("amx-movrs: {:?}", is_x86_feature_detected!("amx-movrs"));
 }
diff -Naru rustc-1.97.1-src/library/stdarch/crates/core_arch/src/x86_64/amx.rs rustc-1.97.1-src/library/stdarch/crates/core_arch/src/x86_64/amx.rs
--- rustc-1.97.1-src/library/stdarch/crates/core_arch/src/x86_64/amx.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/library/stdarch/crates/core_arch/src/x86_64/amx.rs	2026-07-23 17:00:08.703170288 +0200
@@ -358,30 +358,6 @@
     tileloaddrst164(DST as i8, base, stride);
 }
 
-/// Perform matrix multiplication of two tiles a and b, containing packed single precision (32-bit)
-/// floating-point elements, which are converted to TF32 (tensor-float32) format, and accumulate the
-///  results into a packed single precision tile.
-/// For each possible combination of (row of a, column of b), it performs
-///  - convert to TF32
-///  - multiply the corresponding elements of a and b
-///  - accumulate the results into the corresponding row and column of dst using round-to-nearest-even
-/// rounding mode.
-/// Output FP32 denormals are always flushed to zero, input single precision denormals are always
-/// handled and *not* treated as zero.
-#[inline]
-#[rustc_legacy_const_generics(0, 1, 2)]
-#[target_feature(enable = "amx-tf32")]
-#[cfg_attr(
-    all(test, not(target_vendor = "apple")),
-    assert_instr(tmmultf32ps, DST = 0, A = 1, B = 2)
-)]
-#[unstable(feature = "x86_amx_intrinsics", issue = "126622")]
-pub unsafe fn _tile_mmultf32ps<const DST: i32, const A: i32, const B: i32>() {
-    static_assert_uimm_bits!(DST, 3);
-    static_assert_uimm_bits!(A, 3);
-    static_assert_uimm_bits!(B, 3);
-    tmmultf32ps(DST as i8, A as i8, B as i8);
-}
 
 /// Moves a row from a tile register to a zmm register, converting the packed 32-bit signed integer
 /// elements to packed single-precision (32-bit) floating-point elements.
@@ -619,8 +595,6 @@
     fn tileloaddrs64(dst: i8, base: *const u8, stride: usize);
     #[link_name = "llvm.x86.tileloaddrst164"]
     fn tileloaddrst164(dst: i8, base: *const u8, stride: usize);
-    #[link_name = "llvm.x86.tmmultf32ps"]
-    fn tmmultf32ps(dst: i8, a: i8, b: i8);
     #[link_name = "llvm.x86.tcvtrowd2ps"]
     fn tcvtrowd2ps(tile: i8, row: u32) -> f32x16;
     #[link_name = "llvm.x86.tcvtrowd2psi"]
@@ -1454,30 +1428,4 @@
         }
     }
 
-    #[simd_test(enable = "amx-tf32")]
-    fn test_tile_mmultf32ps() {
-        unsafe {
-            _init_amx();
-            let a: [[f32; 16]; 16] = array::from_fn(|i| [i as _; _]);
-            let b: [[f32; 16]; 16] = [array::from_fn(|j| j as _); _];
-            let mut res = [[0.0; 16]; 16];
-
-            let mut config = __tilecfg::default();
-            config.palette = 1;
-            (0..=2).for_each(|i| {
-                config.colsb[i] = 64;
-                config.rows[i] = 16;
-            });
-            _tile_loadconfig(config.as_ptr());
-            _tile_zero::<0>();
-            _tile_loadd::<1>(a.as_ptr().cast(), 64);
-            _tile_loadd::<2>(b.as_ptr().cast(), 64);
-            _tile_mmultf32ps::<0, 1, 2>();
-            _tile_stored::<0>(res.as_mut_ptr().cast(), 64);
-            _tile_release();
-
-            let expected = array::from_fn(|i| array::from_fn(|j| 16.0 * i as f32 * j as f32));
-            assert_eq!(res, expected);
-        }
-    }
 }
diff -Naru rustc-1.97.1-src/library/stdarch/crates/stdarch-verify/tests/x86-intel.rs rustc-1.97.1-src/library/stdarch/crates/stdarch-verify/tests/x86-intel.rs
--- rustc-1.97.1-src/library/stdarch/crates/stdarch-verify/tests/x86-intel.rs	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/library/stdarch/crates/stdarch-verify/tests/x86-intel.rs	2026-07-23 17:00:08.706507960 +0200
@@ -303,7 +303,7 @@
             }
 
             // FIXME: these have not been added to Intrinsics Guide yet
-            if ["amx-avx512", "amx-fp8", "amx-movrs", "amx-tf32", "movrs"]
+            if ["amx-avx512", "amx-fp8", "amx-movrs", "movrs"]
                 .iter()
                 .any(|f| feature.contains(f))
             {
diff -Naru rustc-1.97.1-src/tests/ui/check-cfg/target_feature.stderr rustc-1.97.1-src/tests/ui/check-cfg/target_feature.stderr
--- rustc-1.97.1-src/tests/ui/check-cfg/target_feature.stderr	2026-07-14 17:13:08.000000000 +0200
+++ rustc-1.97.1-src/tests/ui/check-cfg/target_feature.stderr	2026-07-23 17:00:08.700042319 +0200
@@ -27,7 +27,6 @@
 `amx-fp8`
 `amx-int8`
 `amx-movrs`
-`amx-tf32`
 `amx-tile`
 `apxf`
 `atomics`
