diff -up rustc-1.50.0-src/compiler/rustc_codegen_llvm/src/llvm_util.rs.omv~ rustc-1.50.0-src/compiler/rustc_codegen_llvm/src/llvm_util.rs
--- rustc-1.50.0-src/compiler/rustc_codegen_llvm/src/llvm_util.rs.omv~	2021-03-22 02:16:15.364390738 +0100
+++ rustc-1.50.0-src/compiler/rustc_codegen_llvm/src/llvm_util.rs	2021-03-22 02:17:37.903637441 +0100
@@ -177,6 +177,13 @@ pub fn print_version() {
     }
 }
 
+pub fn get_version() -> (u32, u32, u32) {
+    // Can be called without initializing LLVM
+    unsafe {
+        (llvm::LLVMRustVersionMajor(), llvm::LLVMRustVersionMinor(), 0)
+    }
+}
+
 pub fn get_major_version() -> u32 {
     unsafe { llvm::LLVMRustVersionMajor() }
 }
