diff -up rustc-1.71.0-src/vendor/pkg-config-0.3.25/src/lib.rs.omv~ rustc-1.71.0-src/vendor/pkg-config-0.3.25/src/lib.rs
--- rustc-1.71.0-src/vendor/pkg-config-0.3.25/src/lib.rs.omv~	2023-08-16 14:34:33.021530154 +0200
+++ rustc-1.71.0-src/vendor/pkg-config-0.3.25/src/lib.rs	2023-08-16 14:36:36.396850060 +0200
@@ -472,6 +472,7 @@ impl Config {
             .targetted_env_var("PKG_CONFIG")
             .unwrap_or_else(|| OsString::from("pkg-config"));
         let mut cmd = Command::new(exe);
+        let mut pkg_added: bool = false;
         if self.is_static(name) {
             cmd.arg("--static");
         }
@@ -492,25 +493,31 @@ impl Config {
         if self.print_system_cflags {
             cmd.env("PKG_CONFIG_ALLOW_SYSTEM_CFLAGS", "1");
         }
-        cmd.arg(name);
         match self.min_version {
             Bound::Included(ref version) => {
                 cmd.arg(&format!("{} >= {}", name, version));
+                pkg_added = true;
             }
             Bound::Excluded(ref version) => {
                 cmd.arg(&format!("{} > {}", name, version));
+                pkg_added = true;
             }
             _ => (),
         }
         match self.max_version {
             Bound::Included(ref version) => {
                 cmd.arg(&format!("{} <= {}", name, version));
+                pkg_added = true;
             }
             Bound::Excluded(ref version) => {
                 cmd.arg(&format!("{} < {}", name, version));
+                pkg_added = true;
             }
             _ => (),
         }
+        if ! pkg_added {
+            cmd.arg(name);
+        }
         cmd
     }
 
diff -up rustc-1.71.0-src/vendor/pkg-config/src/lib.rs.omv~ rustc-1.71.0-src/vendor/pkg-config/src/lib.rs
--- rustc-1.71.0-src/vendor/pkg-config/src/lib.rs.omv~	2023-08-16 14:27:11.798766119 +0200
+++ rustc-1.71.0-src/vendor/pkg-config/src/lib.rs	2023-08-16 14:34:11.885303636 +0200
@@ -479,6 +479,7 @@ impl Config {
             .targetted_env_var("PKG_CONFIG")
             .unwrap_or_else(|| OsString::from("pkg-config"));
         let mut cmd = Command::new(exe);
+        let mut pkg_added: bool = false;
         if self.is_static(name) {
             cmd.arg("--static");
         }
@@ -499,25 +500,31 @@ impl Config {
         if self.print_system_cflags {
             cmd.env("PKG_CONFIG_ALLOW_SYSTEM_CFLAGS", "1");
         }
-        cmd.arg(name);
         match self.min_version {
             Bound::Included(ref version) => {
                 cmd.arg(&format!("{} >= {}", name, version));
+                pkg_added = true;
             }
             Bound::Excluded(ref version) => {
                 cmd.arg(&format!("{} > {}", name, version));
+                pkg_added = true;
             }
             _ => (),
         }
         match self.max_version {
             Bound::Included(ref version) => {
                 cmd.arg(&format!("{} <= {}", name, version));
+                pkg_added = true;
             }
             Bound::Excluded(ref version) => {
                 cmd.arg(&format!("{} < {}", name, version));
+                pkg_added = true;
             }
             _ => (),
         }
+        if ! pkg_added {
+            cmd.arg(name);
+        }
         cmd
     }
 
