diff -up rustc-1.80.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs.2~ rustc-1.80.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs
--- rustc-1.80.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs.2~	2006-07-24 03:21:28.000000000 +0200
+++ rustc-1.80.0-src/compiler/rustc_codegen_ssa/src/back/linker.rs	2024-08-02 00:26:42.205247951 +0200
@@ -768,6 +768,7 @@ impl<'a> Linker for GccLinker<'a> {
             self.linker_arg("--nxcompat");
         } else if self.is_gnu {
             self.linker_args(&["-z", "noexecstack"]);
+            self.linker_args(&["-z", "nostart-stop-gc"]);
         }
     }
 
@@ -1252,6 +1253,7 @@ impl<'a> Linker for WasmLd<'a> {
 
     fn gc_sections(&mut self, _keep_metadata: bool) {
         self.cmd.arg("--gc-sections");
+        self.cmd.arg("-z,nostart-stop-gc");
     }
 
     fn no_gc_sections(&mut self) {
